src/app/courses/components/courses-item/courses-item.component.ts

```diff
@@ -1,15 +1,21 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ICourseItem } from '../../models/course-item.model';
 
 @Component({
   selector: 'app-courses-item',
   templateUrl: './courses-item.component.html',
-  styleUrls: ['./courses-item.component.css'],
+  styleUrls: ['./courses-item.component.less'],
 })
-export class CoursesItemComponent implements OnInit {
+export class CoursesItemComponent {
+  @Input() courseItem: ICourseItem;
+  @Output() delete = new EventEmitter<number | string>();
 
-  constructor() { }
-
-  ngOnInit() {
+  onDelete() {
+    this.delete.emit(this.courseItem.id);
+  }
+  getformattedDuration(): string {
+    const h = Math.floor(this.courseItem.duration / 60);
+    const m = this.courseItem.duration % 60;
+    return h + 'h' + m + 'min';
```

Comment:
Please use full words for variable names, even for obvious ones like these. One letter variable name is a serious code smell. And template string es6 syntax should be used instead of string concatenation - it's more elegant and expressive.